Frontend/android-native/mobility-app/src/main/java/in/juspay/mobility/app/carousel/VPAdapter.java

```diff
@@ -29,27 +56,188 @@ public ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
+            // imageView Config ------------------------------------------
+            if ((viewPagerItem.contentType).equals("IMAGE")){
+                holder.imageView.setImageResource(viewPagerItem.imageID);
+                holder.imageView.getLayoutParams().height = (int) (imageConfig.getInt("height") * density);
+                GradientDrawable gradientDrawable = new GradientDrawable();
+                gradientDrawable.setShape(GradientDrawable.RECTANGLE);
+                holder.imageView.setVisibility(View.VISIBLE);
+                gradientDrawable.setCornerRadii(new float[] {20, 20, 20, 20, 0,0,0,0});
+                gradientDrawable.setColor(Color.parseColor(viewPagerItem.imageConfig.getString("bgColor")));
+                holder.imageView.setBackground(gradientDrawable);
+                holder.video.setVisibility(View.GONE);
+            }
+            else {
+                embedYoutubeVideo(context, (viewPagerItem.videoData).toString(), String.valueOf(R.id.videoViewLinearLayout), "PLAY", holder.video);
+                holder.video.setVisibility(View.VISIBLE);
+                holder.imageView.setVisibility(View.GONE);
+            }
+
+            // Heading text Config ------------------------------------------
+            holder.tvHeading.setTextSize(titleConfig.getInt("textSize"));
+            holder.tvHeading.setTextColor(Color.parseColor(titleConfig.getString("textColor")));
+            holder.tvHeading.setText(titleConfig.getString("text"));
+            ViewGroup.MarginLayoutParams titleLayoutParams = (ViewGroup.MarginLayoutParams) holder.tvHeading.getLayoutParams();
+            titleLayoutParams.setMargins(titleMargin.getInt("left"), titleMargin.getInt("top"), titleMargin.getInt("right"), titleMargin.getInt("bottom"));
+            holder.tvHeading.setLayoutParams(titleLayoutParams);
+            holder.tvHeading.setGravity(getGravity(titleGravity));
 
+            // Description text Config ---------------------------------------
+            holder.tvDesc.setText(Html.fromHtml(descriptionConfig.getString("text")));
+            if(descriptionConfig.getString("text").equals(""))
+            {
+                holder.tvDesc.setVisibility(View.GONE);
+            }
+            holder.tvDesc.setTextSize(descriptionConfig.getInt("textSize"));
+            holder.tvDesc.setTextColor(Color.parseColor(descriptionConfig.getString("textColor")));
+            ViewGroup.MarginLayoutParams descLayoutParams = (ViewGroup.MarginLayoutParams) holder.tvDesc.getLayoutParams();
+            descLayoutParams.setMargins(margin.getInt("left") * 2, margin.getInt("top"), margin.getInt("right"), margin.getInt("bottom"));
+            holder.tvDesc.setLayoutParams(descLayoutParams);
+            holder.tvDesc.setGravity(getGravity(descriptionGravity));
+            holder.parentLinearLayout.setGravity(carouselGravity);
+          }
+            catch (Exception e){
```

Comment:
What will happen if it comes to Exception case
Will holder be there without any view ? Please check this case, and add an event